src/components/layout/header/Header.tsx

```diff
@@ -1,14 +1,16 @@
 import React, { useCallback, useEffect, useState } from 'react'
-import classNames from 'classnames'
+import Link from 'next/link'
 import Image from 'next/image'
 import { useRouter } from 'next/router'
-import Link from 'next/link'
+import classNames from 'classnames'
 import { useMediaType } from '../../../hook/useMediaType'
-import type { Route } from '../../../routes'
-import { headerRoutes } from '../../../routes'
+import { menu } from '../../../routes'
+import type { Route, SubRoute } from '../../../routes'
 
 type MenuProps = Readonly<{
   routerPath: string | null
+  selectedMenu?: Route | null
+  subMenu?: SubRoute[]
```

Comment:
I would have dissociated the props used by the desktop / mobile menus and the `SubMenuDesktop` component instead of having optional properties that are not used.